cli/src/commands/content/updateChannel.ts

```diff
@@ -92,8 +79,18 @@ export default class UpdateChannelCommand extends UploadCommandBase {
     const meta = asValidatedMetadata(ChannelMetadata, channelInput)
     const { collaborators, coverPhotoPath, avatarPhotoPath } = channelInput
 
-    if (collaborators !== undefined && !this.isChannelOwner(channel, actor)) {
-      this.error("Only channel owner is allowed to update channel's collaborators!", { exit: ExitCodes.AccessDenied })
+    if (
+      collaborators !== undefined &&
+      !this.isChannelOwner(channel, actor) &&
+      !this.isCollaboratorWithRequiredPermission(channel, actor, 'ManageChannelCollaborators')
```

Comment:
I think it makes sense to just generally rework the context & permissions in the CLI, because:
- For curator channels we now need to introduce `Curator` context (while only `Owner` / `Collaborator` is currently available)
- For both `Curator` and `Collaborator` context we need to verify `ChannelAgentPermissions`. In case of `updateChannel` there are 3 different permissions that need to be taken into account: `UpdateChannelMetadata` (in case metadata is changed), `ManageNonVideoChannelAssets` (in case assets are updated), `ManageChannelCollaborators` (in case collaborators are updated). For `ManageChannelCollaborators` ideally you'd also check whether the affected collaborators have lower permissions than the Curator/Collaborator performing the update (see the comments in https://github.com/Joystream/joystream/blob/carthage/runtime-modules/content/src/types.rs#L134). For other actions like `createVideo` / `updateVideo` / `deleteVideo` etc. you also need to check different permissions depending on the parameters.

Reply:
Addressed in [965e1d1](https://github.com/Joystream/joystream/pull/4084/commits/965e1d14433b2f9d5ab9cea37292026a3a2a8304)

1. Isn't for [CuratorGroup](https://github.com/zeeshanakram3/joystream/blob/965e1d14433b2f9d5ab9cea37292026a3a2a8304/cli/src/base/ContentDirectoryCommandBase.ts#L97) owned channels both `Lead` & `Curator` contexts are already captured by `Owner` context? So not sure if introducing `Curator` context is necessary.
2. Since support for NFT actions doesn't exist in CLI, including in `createVideo`/`updateVideo` as these commands just set `autoIssueNft`  to _null_. Hence, `ManageVideoNfts` permission check is missing from both these commands as of now